### PR TITLE
Add KvbAppFilter event group unit tests and minor fixes

### PR DESCRIPTION
**Pull request by shgandhi**
This PR adds KvbAppFilter event group unit tests along with two TRS fixes. Please see individual commits for more detail.

### CODE REVIEW

File: kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp [hunk 1]
```diff
@@ -422,7 +422,7 @@ std::optional<kvbc::categorization::ImmutableInput> KvbAppFilter::getBlockEvents
 kvbc::categorization::EventGroup KvbAppFilter::getEventGroup(kvbc::EventGroupId event_group_id, std::string &cid) {
   LOG_DEBUG(logger_, "getEventGroup " << event_group_id << " for " << client_id_);
   // get global_event_group_id corresponding to trid_event_group_id
-  const auto key = client_id_ + concordUtils::toBigEndianStringBuffer(event_group_id);
+  const auto key = client_id_ + "#" + concordUtils::toBigEndianStringBuffer(event_group_id);
```
Comment (robem) [inline kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp:425]: This needs a comment otherwise it is an arbitrary `#` :)

Comment (shgandhi) [reply kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp:441]: Updated.

File: kvbc/test/kvbc_app_filter/kvbc_filter_test.cpp [hunk 2]
```diff
@@ -151,6 +198,32 @@ class FakeStorage : public concord::kvbc::IReader {
     }
   }
 
+  // Dummy method to fill DB for testing, each client id can watch only the
+  // event group id that corresponds to their client id.
+  void fillWithEventGroupData(EventGroupId num_of_egs) {
```
Comment (robem) [inline kvbc/test/kvbc_app_filter/kvbc_filter_test.cpp:203]: Number of eggs 🥚  :)

File: kvbc/test/kvbc_app_filter/kvbc_filter_test.cpp [hunk 3]
```diff
@@ -211,6 +284,34 @@ TEST(kvbc_filter_test, kvbfilter_update_success) {
...
+  event2.data = val_trid2;
+  event2.tags = {"0"};
+  event_group.events.emplace_back(event1);
+  event_group.events.emplace_back(event2);
+
+  const auto &filtered = kvb_filter.filterEventGroupUpdate({eg_id, event_group});
+
+  EXPECT_EQ(filtered.event_group_id, eg_id);
+  EXPECT_EQ(filtered.event_group.events.size(), 1);
+  for (auto &event : filtered.event_group.events) {
```
Comment (robem) [inline kvbc/test/kvbc_app_filter/kvbc_filter_test.cpp:310]: Nit: We don't need a for-loop for a single event.

**Review by robem — APPROVED** [review]
Lgtm thanks! 💪

File: kvbc/test/kvbc_app_filter/kvbc_filter_test.cpp (hunk 3 above)
Comment (shgandhi) [reply kvbc/test/kvbc_app_filter/kvbc_filter_test.cpp:310]: Thanks for catching that. Will update in this PR if it fails (4th run 🤞 ), else will add the change in another PR.

**Pull request merged**